### PR TITLE
Add file extension to link & clean up code

### PR DESCRIPTION
**Pull request by spikespaz**
I noticed some things that apparently your linter didn't catch. The quotes were a mix of single and double, your parentheses and indentation were strange, and you were using `function() {}` and `() => {}`. Your function call path was messy and ready to break at the slightest change.

I fixed these issues to make the code more consistent as well as removing the sample comments that came from wherever you copied the original file from.

Interestingly enough, I couldn't `require("vscode")`, even though I can in every other extension, ever. So that part is not like the others because I'm lazy and I gave up.

Additionally, I made sure that Hastebin uses the correct syntax highlighting by adding the file extension of the open document's filename to the end of the assembled Hastebin URL. This fixed some issues that I had with D code.

You were missing the "license" field in your `package.json`, so I fixed that too.

Great extension, thanks!

### CODE REVIEW

**Comment by HamzaAnis** [comment]
Thank you very much for the PR and cleaning up the code. I was using the default linter however good catch for this. I'll test it soon and will merge it.

File: src/extension.ts [hunk 1]
```diff
@@ -1,68 +1,71 @@
-'use strict';
-// The module 'vscode' contains the VS Code extensibility API
```
Comment (HamzaAnis) [inline src/extension.ts:2]: I generated the starting code using [Yo code](https://github.com/Microsoft/vscode-generator-code) so these were here from start.

Comment (spikespaz) [reply src/extension.ts:2]: I see. Well, `"use strict";` is still there and the comments didn't add much, I hope you are fine with them being gone.

File: src/extension.ts [hunk 2]
```diff
@@ -1,68 +1,71 @@
...
+// Read the text from the current window's open document, or selection
+function readDocument(): string {
+    let code;
+
     if (!vscode.window.activeTextEditor) {
         return;
-    } else if (!vscode.window.activeTextEditor.selection.isEmpty) {
-        code = vscode.window.activeTextEditor.document.getText(vscode.window.activeTextEditor.selection);
-    } else {
+    } else if (vscode.window.activeTextEditor.selection.isEmpty) {
```
Comment (HamzaAnis) [inline src/extension.ts:62]: Easier to read now. 👍

File: src/extension.ts [hunk 3]
```diff
@@ -1,68 +1,71 @@
...
+
     if (!content) {
+        vscode.window.showWarningMessage("Could not upload empty document.")
+
         return;
-    } else {
-        uploadCode(content);
     }
+
+    uploadText(content).then((response: any) => {
```
Comment (HamzaAnis) [inline src/extension.ts:27]: This is more convenient. ✔

File: src/extension.ts (hunk 1 above)
Comment (HamzaAnis) [reply src/extension.ts:2]: Yes I am absolutely fine. You did a good job improving it and fixing the mistakes.

**Review by HamzaAnis — APPROVED** [review]

**Pull request merged**